src/k8s_client/k8s_client.go

```diff
@@ -598,12 +598,16 @@ func (c *k8sClient) CreateEvent(namespace, name, message, component string) (*v1
 }
 
 func (c *k8sClient) GetCSVFromSubscription(namespace string, name string) (string, error) {
-	result, err := c.olmClient.Subscriptions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	subscription, err := c.olmClient.Subscriptions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 	if err != nil {
 		c.log.WithError(err).Warnf("Failed to get subscription %s in %s", name, namespace)
 		return "", err
 	}
-	return result.Status.CurrentCSV, nil
+	if subscription.Status.CurrentCSV == "" {
+		c.log.WithError(err).Warnf("subscription CurrentCSV is empty, subscription status: %+v", subscription.Status)
```

Comment:
return error, it should not happen i assume ?

Reply:
It does happen, while we wait

Reply:
so not better to return error in that case?

Reply:
hmm not sure. Depends on how the users of this function behave

Reply:
It get called in operatorHandler `IsInitialized` and `GetStatus` in both cases it's eventually wrapped in a loop.
I don't want to change behavior here. just add logging so it's clear that the CSV is empty and whats the subscription status